core/__tests__/AsyncStorage.test.ts

```diff
@@ -0,0 +1,121 @@
+import AsyncStorage from '../src/AsyncStorage';
+
+class StorageMock implements IStorageBackend<any> {
+  getSingle = jest.fn();
+  setSingle = jest.fn();
+  getMany = jest.fn();
+  setMany = jest.fn();
+  removeSingle = jest.fn();
+  removeMany = jest.fn();
+  getKeys = jest.fn();
+  dropStorage = jest.fn();
+}
+
+describe('AsyncStorage', () => {
+  const mockedStorage = new StorageMock();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('main API', () => {
+    it('handles basic set/read/remove calls', async () => {
+      const as = new AsyncStorage(mockedStorage, {
+        logger: false,
+        errorHandler: false,
+      });
+
+      const key = 'myKey';
+      const value = {
+        name: 'Jerry',
+      };
+
+      // get
+      await as.get(key);
+      expect(mockedStorage.getSingle).toBeCalledWith(key, null);
+
+      //set
+      await as.set(key, value);
+      expect(mockedStorage.setSingle).toBeCalledWith(key, value, null);
+
+      // remove
+      await as.remove(key);
+      expect(mockedStorage.removeSingle).toBeCalledWith(key, null);
+    });
+
+    it('handles basic multi set/read/remove calls', async () => {
+      const keys = ['key1', 'key2', 'key3'];
+      const keyValues = [{key1: 'value1'}, {key2: 'value2'}, {key3: 'value3'}];
+
+      const as = new AsyncStorage(mockedStorage, {
+        logger: false,
+        errorHandler: false,
+      });
+
+      // set
+      await as.setMultiple(keyValues);
+      expect(mockedStorage.setMany).toBeCalledWith(keyValues, null);
+
+      // get
+      await as.getMultiple(keys);
+      expect(mockedStorage.getMany).toBeCalledWith(keys, null);
+
+      // remove
+      await as.removeMultiple(keys);
+      expect(mockedStorage.removeMany).toBeCalledWith(keys, null);
+    });
+
+    it('handles getKeys/clearStorage/instance calls', async () => {
+      const asyncStorage = new AsyncStorage(mockedStorage, {
+        logger: false,
+        errorHandler: false,
+      });
+
+      // instance
+      expect(asyncStorage.instance()).toBe(mockedStorage);
```

Comment:
is this a part of your public API? `asyncStorage` is already an instance of a class, so this is quite unintuitive

Reply:
Yes, it is. Some storages might come with other functionalities than `AsyncStorage` requires, so `instance` gives direct access to the storage (and to said functionalities)

Reply:
Hm, this is dangerous. All functionalities should be behind a facade of your API, don't you think?

Reply:
That'd be ideal, but we cannot predict all possibilities, right?

Reply:
I'd be cautious about that, this is what Enzyme does that sooner or later made everyone using this API because guides and blog post were made. I'd remove this and think about expanding API when necessary

Reply:
Going to leave it like that for now, with a note to re-think it later.